taxonomy/utils.py

```diff
@@ -63,7 +63,8 @@ def get_whitelisted_serialized_skills(key_or_uuid, product_type=ProductTypes.Cou
     if cached_response.is_found:
         return cached_response.value
 
-    product_skills = get_whitelisted_product_skills(key_or_uuid, product_type)
+    whitelisted_product_skills = get_whitelisted_product_skills(key_or_uuid, product_type)
+    product_skills = whitelisted_product_skills.prefetch_related('skill__category', 'skill__subcategory')
```

Comment:
Can we add a quick unit test to see how many queries are happening? you can add count reduction in PR description.